app/controllers/application_controller.rb

```diff
@@ -1,4 +1,14 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
   protect_from_forgery with: :exception
+
+  def handle_service_result(result, on_success, on_failure)
```

Comment:
@LucasAnderson1984 and this. If you look at `truck_scale`, `dispatch`, or `cantina` you should be able to see how I achieved this with a concern and without poo'ing up the `ApplicationController`.